modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/EnrichmentManager.scala

```diff
@@ -93,42 +92,51 @@ object EnrichmentManager {
 
   private val atomicSchema: SchemaKey = SchemaKey("com.snowplowanalytics.snowplow", "atomic", "jsonschema", SchemaVer.Full(1, 0, 0))
 
+
   private def validateEnriched[F[_]: Clock: Monad: RegistryLookup](
     enriched: EnrichedEvent,
     raw: RawEvent,
     processor: Processor,
     client: Client[F, Json]
-  ): EitherT[F, BadRow, Unit] =
-      enriched.toShreddedEvent
-        .leftMap(err =>
-          EnrichmentManager.buildEnrichmentFailuresBadRow(
-            NonEmptyList.one(
-              FailureDetails.EnrichmentFailure(Some(EnrichmentInformation(atomicSchema, identifier = "atomic_validator")),
-                                               EnrichmentFailureMessage.Simple(err.getMessage)
+  ): EitherT[F, BadRow, Unit] = enriched.toAtomicEvent
+      .leftMap(err =>
+        EnrichmentManager.buildEnrichmentFailuresBadRow(
+          NonEmptyList(
+            EnrichmentFailure(
+              None,
+              FailureDetails.EnrichmentFailureMessage.Simple(
+                "Critical error during conversion of enriched event to the shredded format"
               )
             ),
-            toPartiallyEnrichedEvent(enriched),
-            toRawEvent(raw),
-            processor
-          )
+            List(EnrichmentFailure(None, FailureDetails.EnrichmentFailureMessage.Simple(err.getMessage)))
+          ),
+          toPartiallyEnrichedEvent(enriched),
+          toRawEvent(raw),
+          processor
         )
-        .toEitherT[F]
-        .flatMap(atomic =>
-          client
-            .check(
-              SelfDescribingData(atomicSchema, shreded)
-            )
-            .leftMap(err =>
-              buildSchemaViolationsBadRow(
-                NonEmptyList.one(
-                  FailureDetails.SchemaViolation.IgluError(atomicSchema, err)
+      )
+      .toEitherT[F]
+      .flatMap(atomic =>
+        client
+          .check(SelfDescribingData(atomicSchema, atomic))
+          .leftMap(err =>
+            EnrichmentManager.buildEnrichmentFailuresBadRow(
+              NonEmptyList(
+                EnrichmentFailure(
+                  None,
+                  FailureDetails.EnrichmentFailureMessage.Simple(
+                    s"Enriched event output validation error against the ${atomicSchema.toSchemaUri}. In most cases" +
+                      " it would be because of oversized fields."
```

Comment:
I would not add `in most cases ...` because it can be misleading about the cause and anyway the error message from the validation should say what the problem is